src/ert/scheduler/scheduler.py

```diff
@@ -107,8 +115,17 @@ def kill_all_jobs(self) -> None:
 
     async def cancel_all_jobs(self) -> None:
         self._cancelled = True
+        await self._cancel_tasks()
+
+    async def _cancel_tasks(self) -> None:
         for task in self._tasks.values():
             task.cancel()
+        with suppress(asyncio.TimeoutError):
+            await asyncio.wait(
+                list(self._tasks.values()),
```

Comment:
Do we need to make it a list, or can we use the view directly?

Reply:
You might be right as the only requirement is that it's Iterable